examples/hackernews/reverse_proxy/haproxy.distributed.cfg

```diff
@@ -15,18 +16,27 @@ frontend www
     # Default to serving the www frontend.
     default_backend www
 
-# Skip streaming backends
-backend follower1
+
+# Skip control backend
+listen control
     mode http
-    http-request set-path /v1/streams/%[path,field(4,/)]
-    server follower1 skip_follower1:8080
-backend follower2
+    bind :8081
+    # Writes to input collections go to leader
+    use-server leader if { path_beg -i /v1/inputs/ }
+    server leader skip_leader:8081 weight 0
+    # Other (stream-creation) requests go to followers
+    balance roundrobin
+    server follower1 skip_follower1:8081
+    server follower2 skip_follower2:8081
+    server follower3 skip_follower3:8081
```

Comment:
FMI, is this sort of repetition something that can be made parametric reasonably easily?

Reply:
There's [`server-template`](https://docs.haproxy.org/3.0/configuration.html#4.2-server-template), but that requires all of the servers to have the same domain name.

My plan here is to use [haproxy maps](https://docs.haproxy.org/3.0/configuration.html#7.3.1-map) to make this elastic, using set-map/del-map to update these mappings on the fly.  but still figuring out how to make that work :)